src/ship/PlayerShipController.cpp

```diff
@@ -687,9 +696,9 @@ void PlayerShipController::PollControls(const float timeStep, int *mouseMotion,
 		m_mouseActive = false;
 	}
 
-	if (InputBindings.primaryFire->IsActive() || (Pi::input->MouseButtonState(SDL_BUTTON_LEFT) && Pi::input->MouseButtonState(SDL_BUTTON_RIGHT))) {
-		//XXX worldview? madness, ask from ship instead
-		m_ship->SetGunState(Pi::game->GetWorldView()->GetActiveWeapon(), 1);
+	if (gunManager &&
+		(InputBindings.primaryFire->IsActive() || (Pi::input->MouseButtonState(SDL_BUTTON_LEFT) && Pi::input->MouseButtonState(SDL_BUTTON_RIGHT)))) {
+		gunManager->SetAllGroupsFiring(true);
```

Comment:
Note that this change will cause weapons in group 0 (by default, forward-facing weapons) to fire when the player is looking backwards and trying to fire weapons in group 1 (by default, rear-facing weapons).

I'm not sure we currently have ships with rear-facing weapons at this time, but I'd appreciate a comment here indicating that support for per-view weapon groups needs to be restored in the future.

Reply:
That works, thank you. In the future I'd like to assign mounts to individual views or "seats" (e.g. crew-manned turrets) but for now we can use the WorldView.

Reply:
Force-pushed this change.